operator/clusterutilities.go

```diff
@@ -422,6 +441,67 @@ func GetReplicaAffinity(clusterLabels, replicaLabels map[string]string) string {
 	return GetAffinity(key, value, operator)
 }
 
+// GetPodAntiAffinity returns the populated pod anti-affinity json that should be attached to
+// the various pods comprising the pg cluster
+func GetPodAntiAffinity(podAntiAffinityType string, clusterName string) string {
+
+	log.Debugf("GetPodAnitAffinity with clusterName=[%s]", clusterName)
+
+	// get the PodAntiAffinity type from the CR parameter (podAntiAffinityType) or from the
+	// pgo.yaml, depending on whether or not either have been set
+	var affinityTypeParam crv1.PodAntiAffinityType
+	if podAntiAffinityType != "" {
+		affinityTypeParam = crv1.PodAntiAffinityType(podAntiAffinityType)
+	} else if Pgo.Cluster.PodAntiAffinity != "" {
+		affinityTypeParam = crv1.PodAntiAffinityType(Pgo.Cluster.PodAntiAffinity)
+	} 
+
+	// verify that the affinity type provided is valid (i.e. 'required' or 'preffered'), and
+	// log an error and return an empty string if not
+	if err := affinityTypeParam.ValidatePodAntiAffinityType(); affinityTypeParam != "" &&
```

Comment:
Same here, shouldn't the `""` be handled by validation?

Reply:
If an empty affinity type is provided, then it will default to preferred once processed by the operator. Therefore, since a empty value is not intended to be a affinity type, it does not need to be validated. Instead, it simply indicates that the default should be applied (specifically by the Operator).